File: problem938.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    int rangeSumBST(TreeNode* root, int L, int R) {
        search(root, L, R);
        return res;
    }
    
    int res = 0;
    
    void search(TreeNode* root, int L, int R){
        if (root==nullptr){
            return;
        } else {
            if (L<=root->val && root->val<=R){
                res += root->val;
            }
            if (L<root->val) search(root->left, L, R);
            if (root->val<R) search(root->right, L, R);
        }
    }
};

File: problem938/problem938.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def rangeSumBST(self, root, L, R):
        self.res = 0

        def search(node):
            if node is not None:
                nv = node.val
                if L <= nv <= R:self.res += node.val
                if L < nv:search(node.left)
                if nv < R:search(node.right)

        search(root)
        return self.res